frontend/src/metabase/writeback/components/ActionCreator/FormCreator/constants.ts

```diff
@@ -0,0 +1,90 @@
+import { t } from "ttag";
+import type { FieldType, InputType } from "metabase-types/api/writeback";
+
+interface FieldOptionType {
+  value: FieldType;
+  name: string;
+}
+
+export const fieldTypes: FieldOptionType[] = [
+  {
+    value: "text",
+    name: t`text`,
+  },
+  {
+    value: "number",
+    name: t`number`,
+  },
+  {
+    value: "date",
+    name: t`date`,
+  },
+  {
+    value: "category",
+    name: t`category`,
+  },
+];
```

Comment:
There is some room for i18n issues here. As `fieldTypes` are defined ones, their `names` will remain in the language you had when you opened Metabase. So if you change the language, you might see those types written in the previous language actually. Ideas on how to resolve that:

1. Wrap it inside a function like `getFieldTypes()` and call it inside the component that was using it
2. Add a function like `getNameForFieldType` that'd return a translated name given the `value` prop

Reply:
Same applies to other `ttag` usages below in this file